src/Handler/Backend/Init/InitBackendHandler.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\StorageDriver\BigQuery\Handler\Backend\Init;
+
+use Keboola\StorageDriver\BigQuery\GCPClientManager;
+use Keboola\StorageDriver\BigQuery\IAmPermissions;
+use Keboola\StorageDriver\Credentials\BigQueryCredentials;
+use Keboola\StorageDriver\Shared\Driver\Exception\Exception;
+use Google\Protobuf\Internal\Message;
+use Keboola\StorageDriver\Command\Backend\InitBackendCommand;
+use Keboola\StorageDriver\Command\Backend\InitBackendResponse;
+use Keboola\StorageDriver\Contract\Driver\Command\DriverCommandHandlerInterface;
+
+final class InitBackendHandler implements DriverCommandHandlerInterface
+{
+    public GCPClientManager $clientManager;
+
+    public function __construct(GCPClientManager $clientManager)
+    {
+        $this->clientManager = $clientManager;
+    }
+
+    /**
+     * @inheritDoc
+     * @param BigQueryCredentials $credentials
+     */
+    public function __invoke(
+        Message $credentials,
+        Message $command,
+        array $features
+    ): ?Message {
+        assert($credentials instanceof BigQueryCredentials);
+        assert($command instanceof InitBackendCommand);
+
+        $foldersClient = $this->clientManager->getFoldersClient($credentials);
+
+        try {
+            $formattedName = $foldersClient->folderName($credentials->getFolderId());
+            $folderPermissions = $foldersClient->testIamPermissions(
+                $formattedName,
+                [
+                    IAmPermissions::RESOURCE_MANAGER_PROJECTS_CREATE
+                ]
+            );
+        } finally {
+            $foldersClient->close();
```

Comment:
když mrkneš na TD driver tak ten má https://github.com/keboola/php-storage-driver-teradata/blob/main/src/TeradataSessionManager.php aby na něco takového nešlo zapomnět. https://github.com/keboola/php-storage-driver-teradata/blob/main/src/TeradataDriverClient.php#L60-L70

Reply:
jj ja som to videl, ale musim to este nejak lepsie vymysliet pretoze BQ ma pre kazdy resource ineho klienta takze ich bude este viac inicializovat v ramci toho managera takze pak musim nejak rozumne volat `close` vyriesim to asi pri vytvarani projektu kde uz bude viac tych klientov